### PR TITLE
Implement harvest action and command

### PR DESCRIPTION
**Pull request by girikuncoro**
Merge #25

### CODE REVIEW

File: app/api/sms/harvest_action.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            raise Exception('harvest command is invalid')
+
+        new = Farm(id=Farm.id(),
+                   district_id=self.district_id,
+                   action=self.CMD,
+                   crop_name=self.harvest,
+                   quantity=self.amount)
+        new.put()
+
+        return new.toJson()
```
Comment (Ragil) [inline app/api/sms/harvest_action.py]: why are we sending the json as the sms response?

Comment (girikuncoro) [inline app/api/sms/harvest_action.py]: wait, so the command action not only stores data, but should also include sms response to the user as well? in this case, the response for valid and invalid SMS are different, right? what else should be included?

Comment (Ragil) [inline app/api/sms/harvest_action.py]: the action should perform everything related to that command.

File: app/api/sms/harvest_action.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def __init__(self, command):
+        super(HarvestAction, self).__init__(command)
+        self.harvest = command.harvest
+        self.amount = command.amount
+        self.district_id = command.district()
+        self.valid = command.valid()
+
+    def execute(self):
+        if not self.valid:
+            raise Exception('harvest command is invalid')
```
Comment (Ragil) [inline app/api/sms/harvest_action.py]: This is pointless. The dispatcher will never dispatch an action if the command is invalid

File: app/api/sms/harvest_action.py (hunk 1 above)
Comment (girikuncoro) [inline app/api/sms/harvest_action.py]: what is everything?

Comment (Ragil) [inline app/api/sms/harvest_action.py]: That is all that is related to this particular action.

Comment (girikuncoro) [inline app/api/sms/harvest_action.py]: Cool, I'll try to fix it tonight!

File: app/api/sms/harvest_action.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        if self.args[1] and self.args[1].isdigit():
+            self.harvest = self.args[0]
+            self.amount = int(self.args[1])
+
+    def valid(self):
+        valid_cmd = any([self.cmd == cmd for cmd in self.VALID_CMDS])
+        return valid_cmd and self.amount and self.harvest
+
+    def district(self):
+        user = User.query(User.phone_number == self.sms.from_number).get()
```
Comment (Ragil) [inline app/api/sms/harvest_action.py]: This is not required. The sms request should already have a user attached to it. https://github.com/Ragil/Communicate-Indonesia/blob/master/app/api/sms/main.py#L45

File: app/api/sms/harvest_action.py (hunk 1 above)
Comment (girikuncoro) [inline app/api/sms/harvest_action.py]: Fixed in #34 . Uh, now I kinda understand, yesterday didn't look at `Dispatcher` in `app.command.base`

Comment (girikuncoro) [inline app/api/sms/harvest_action.py]: I'll continue with #26 and #12

**Pull request merged**